core/base/Module.php

```diff
@@ -188,21 +193,32 @@ public function init()
                 throw new InvalidConfigException(sprintf('The required component "%s" is not registered in the configuration file', $component));
             }
         }
-        
+
         static::onLoad();
     }
 
     /**
-     * Override the default implementation of Yii's getLayoutPath(). If the property `$useAppLayoutPath` is true,.
-     *
-     * the *@app* namespace views will be looked up for view files
+     * Override the default implementation of Yii's getLayoutPath(). If the property `$useAppLayoutPath` is true,
+     * the *@app* namespace views will be looked up for view files.
+     * Else the layout path of the active theme will be used.
      *
      * @return string
      * @see \yii\base\Module::getLayoutPath()
      */
     public function getLayoutPath()
     {
-        if ($this->useAppLayoutPath) {
+        $this->theme = Yii::$app->themeManager->getActiveTheme();
+
+        if ($this->theme){
+            $this->layout = $this->theme->layout;
+    
+            if ($this->useAppLayoutPath) {
+                // @todo Layout inheritance
```

Comment:
+ this part of the code is untested as i can see from Code Climate
+ there is a todo - is this finished?

Reply:
The todo is obsolete.